src/com/facebook/buck/jvm/java/javax/SynchronizedToolProvider.java

```diff
@@ -0,0 +1,29 @@
+package com.facebook.buck.jvm.java.javax;
+
+import javax.tools.JavaCompiler;
+import javax.tools.ToolProvider;
+
+/**
+ * ToolProvider has no synchronization internally, so if we don't synchronize from the
+ * outside we could wind up loading the compiler classes multiple times from different
+ * class loaders.
+ */
+public class SynchronizedToolProvider {
```

Comment:
If this PR is going to be in review for some other fixes you may want to move change related to this class to a separate PR

Reply:
This change was requested from @jkeljo in order to get this PR done. There was no reason for this class prior to the changes related to this PR